### PR TITLE
document cython and numpy install prereqs

### PR DESCRIPTION
**Pull request by xiantra-eve**

### CODE REVIEW

**Comment by xiantra-eve** [comment]
I noticed you wanted to use https://pypi.python.org/pypi/retry but thought you might also appreciate not adding another dependency.  Since errors are infrequent exponential backoff isn't too important, nor is providing a retry limit.  I'm also of a mind that code should not abort itself when the user might want to just get a cup of coffee...

File: publicAPI/crest_utils.py [hunk 1]
```diff
@@ -294,12 +295,20 @@ def fetch_esi_endpoint(
...
-    )
+    while True:
+        # no try-except, catch in caller
+        # done to make logging path easier
+        req = requests.get(
+            esi_url,
+            headers=headers
+        )
+
+        if req.status_code == 502:      # Server Error: Bad Gateway for url
```
Comment (lockefox) [inline publicAPI/crest_utils.py]: potential infinite blocking loop

File: publicAPI/crest_utils.py [hunk 2]
```diff
@@ -294,12 +295,20 @@ def fetch_esi_endpoint(
         'User-Agent': config.get('GLOBAL', 'useragent')
     }
 
-    # no try-except, catch in caller
-    # done to make logging path easier
-    req = requests.get(
-        esi_url,
-        headers=headers
-    )
+    while True:
```
Comment (lockefox) [inline publicAPI/crest_utils.py]: new code requires coverage.  Coverage is:
-- tests that can be run in the unit tests
-- `logger.error()` that alerts humans of issues

This fails both as testable (need a way to magic in 502 errors) and does not alert people if recovery fails

File: publicAPI/crest_utils.py (hunk 1 above)
Comment (xiantra-eve) [inline publicAPI/crest_utils.py]: Yes, as designed.  I don't want to arrive home 12 hrs later to see that the remote host was down for 2 minutes (or 10) and that caused my data pull to fail.  I want it to keep retrying until *I* tell it to stop.  Of course you're the boss.  :)

File: publicAPI/crest_utils.py [hunk 3]
```diff
@@ -247,12 +248,19 @@ def fetch_crest_endpoint(
...
-    )
+    @retry(wait_fixed=2000, stop_max_delay=10000)
+    def fetch_crest_endpoint_get():
+        # no try-except, catch in caller
+        # done to make logging path easier
+        req = requests.get(
+            crest_url,
+            headers=headers
+        )
+        if req.status_code == 502:          # Server Error: Bad Gateway for url
```
Comment (lockefox) [inline publicAPI/crest_utils.py]: not needed, L264 covers raise requirement

Comment (xiantra-eve) [inline publicAPI/crest_utils.py]: Ok, removing the if will cover all errors (even syntax ;)

Comment (lockefox) [inline publicAPI/crest_utils.py]: unit_tests should cover us for badly-formed requests, and weekly cron will fail if ESI API updates.  Will still raise (same failed request) log and alert if things go wrong

**Pull request merged**